pkgs/tools/virtualization/ovftool/default.nix

```diff
@@ -1,149 +1,253 @@
-                    ovftoolSystems.${system}
-                  else throw "System '${system}' is unsupported by ovftool";
+  ovftoolSystem = ovftoolSystems.${stdenv.system} or (throw "unsupported system ${stdenv.system}");
 
-  ovftoolSource = if builtins.hasAttr system ovftoolBundles then
-                    ovftoolBundles.${system}
-                  else
-                    fetchurl {
-                      inherit (ovftoolSystem) name url hash;
-                    };
 in
-stdenv.mkDerivation rec {
+stdenv.mkDerivation {
   pname = "ovftool";
-  inherit version;
+  inherit (ovftoolSystem) version;
 
-  src = ovftoolSource;
+  src = fetchurl {
+    inherit (ovftoolSystem) name url hash;
+  };
 
   buildInputs = [
-    glibc
-    libxcrypt-legacy
     c-ares
     expat
     icu60
+    libiconv
+    libxcrypt-legacy
     xercesc
     zlib
+  ] ++ lib.optionals stdenv.isLinux [
+    glibc
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.Libsystem
+    libxml2
   ];
 
-  nativeBuildInputs = [ autoPatchelfHook makeWrapper unzip ];
-
-  preferLocalBuild = true;
-
-  sourceRoot = ".";
+  nativeBuildInputs = [ unzip makeWrapper ]
+    ++ lib.optionals stdenv.isLinux [ autoPatchelfHook ];
 
-  unpackPhase = ovftoolSystem.unpackPhase;
-
-  # Expects a directory named 'ovftool' containing the ovftool install.
-  # Based on https://aur.archlinux.org/packages/vmware-ovftool/
-  # with the addition of a libexec directory and a Nix-style binary wrapper.
   installPhase = ''
     runHook preInstall
-    if [ -d ovftool ]; then
-      # Ensure we're in the staging directory
-      cd ovftool
+
+    # Based on https://aur.archlinux.org/packages/vmware-ovftool/
+    # with the addition of a libexec directory and a Nix-style binary wrapper.
+
+    # Almost all libs in the package appear to be VMware proprietary except for
+    # libgoogleurl and libcurl. The rest of the libraries that the installer
+    # extracts are omitted here, and provided in buildInputs. Since libcurl
+    # depends on VMware's OpenSSL, both libs are still used.
+    # FIXME: Replace libgoogleurl? Possibly from Chromium?
+    # FIXME: Tell VMware to use a modern version of OpenSSL. As of ovftool
+    # v4.6.2 ovftool uses openssl-1.0.2zh which in seems to be the extended
+    # support LTS release: https://www.openssl.org/support/contracts.html
+
+    # Remove packaged libs that are linked during fixup. Darwin dylibs are
+    # under lib in the zip.
+    rm -f \
+      libcares.so.2 \
+      libexpat.so \
+      libgcc_s.so.1 \
+      libicudata.so.60 \
+      libicuuc.so.60 \
+      libstdc++.so.6 \
+      libxerces-c-3.2.so \
+      libz.so.1 \
+      lib/libcares.2.dylib \
+      lib/libexpat.dylib \
+      lib/libicudata.60.2.dylib \
+      lib/libicuuc.60.2.dylib \
+      lib/libxerces-c-3.2.dylib \
+
+    # Install all remaining libs
+    install -m 755 -d "$out/lib"
+    install -m 644 -t "$out/lib" *.so *.so.* "lib/"*.dylib
```

Comment:
How any are those? If we just want to install a few, we maybe want to list those instead the ones we delete.

Reply:
Yeah we had it like this before. It was something like 

```
    install -m 644 -t "$out/lib/${pname}" \
      libgoogleurl.so.59 \
      libssoclient.so \
      libvim-types.so libvmacore.so libvmomi.so \
      libcurl.so.4 libcrypto.so.1.0.2 libssl.so.1.0.2
```

So it may be better to just list them in an allowlist fashion since there are as many being installed as being deleted.

Reply:
I think tracking the installed libraries is easier in the long run and we don#t miss deleting some.

Also if this is using openssl 1.0.2 we should mark the package as insecure.

Reply:
> Yeah we had it like this before. It was something like
> 
> ```
>     install -m 644 -t "$out/lib/${pname}" \
>       libgoogleurl.so.59 \
>       libssoclient.so \
>       libvim-types.so libvmacore.so libvmomi.so \
>       libcurl.so.4 libcrypto.so.1.0.2 libssl.so.1.0.2
> ```
> 
> So it may be better to just list them in an allowlist fashion since there are as many being installed as being deleted.

I'll make the change back to the explicit inclusion of libs.

Reply:
> I think tracking the installed libraries is easier in the long run and we don#t miss deleting some.
> 
> Also if this is using openssl 1.0.2 we should mark the package as insecure.

Probably. I'll mark it as insecure. But I do see it using the paid for extended support version of openssl-1.0.2zh. This has some security patches beyond what the open source 1.0.2 version has, though I would assume that they are not all patched.